error[E0277]: the trait bound `Q: FromParam<'_>` is not satisfied
 --> $DIR/route-type-errors.rs:8:7
  |
8 | fn f0(foo: Q) {} //~ ERROR FromParam
  |       ^^^^^^ the trait `FromParam<'_>` is not implemented for `Q`
  |
  = note: required by `from_param`

error[E0277]: the trait bound `Q: FromSegments<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:11:7
   |
11 | fn f1(foo: Q) {} //~ ERROR FromSegments
   |       ^^^^^^ the trait `FromSegments<'_>` is not implemented for `Q`
   |
   = note: required by `from_segments`

error[E0277]: the trait bound `Q: FromFormValue<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:14:7
   |
14 | fn f2(foo: Q) {} //~ ERROR FromFormValue
   |       ^^^^^^ the trait `FromFormValue<'_>` is not implemented for `Q`
   |
   = note: required by `from_form_value`

error[E0277]: the trait bound `Q: FromQuery<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:17:7
   |
17 | fn f3(foo: Q) {} //~ ERROR FromQuery
   |       ^^^^^^ the trait `FromQuery<'_>` is not implemented for `Q`
   |
   = note: required by `from_query`

error[E0277]: the trait bound `Q: FromDataSimple` is not satisfied
  --> $DIR/route-type-errors.rs:20:7
   |
20 | fn f4(foo: Q) {} //~ ERROR FromData
   |       ^^^^^^ the trait `FromDataSimple` is not implemented for `Q`
   |
   = note: required because of the requirements on the impl of `FromData<'_>` for `Q`

error[E0277]: the trait bound `Q: FromRequest<'_, '_>` is not satisfied
  --> $DIR/route-type-errors.rs:23:7
   |
23 | fn f5(a: Q, foo: Q) {}
   |       ^^^^ the trait `FromRequest<'_, '_>` is not implemented for `Q`
   |
   = note: required by `from_request`

error[E0277]: the trait bound `Q: FromParam<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:23:13
   |
23 | fn f5(a: Q, foo: Q) {}
   |             ^^^^^^ the trait `FromParam<'_>` is not implemented for `Q`
   |
   = note: required by `from_param`

error[E0277]: the trait bound `Q: FromRequest<'_, '_>` is not satisfied
  --> $DIR/route-type-errors.rs:28:7
   |
28 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |       ^^^^ the trait `FromRequest<'_, '_>` is not implemented for `Q`
   |
   = note: required by `from_request`

error[E0277]: the trait bound `Q: FromParam<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:28:13
   |
28 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |             ^^^^^^ the trait `FromParam<'_>` is not implemented for `Q`
   |
   = note: required by `from_param`

error[E0277]: the trait bound `Q: FromParam<'_>` is not satisfied
  --> $DIR/route-type-errors.rs:28:34
   |
28 | fn f6(a: Q, foo: Q, good: usize, bar: Q) {}
   |                                  ^^^^^^ the trait `FromParam<'_>` is not implemented for `Q`
   |
   = note: required by `from_param`
